python/cudf/cudf/tests/test_column.py

```diff
@@ -367,6 +367,32 @@ def test_as_column_buffer(data, expected):
     assert_eq(cudf.Series(actual_column), cudf.Series(expected))
 
 
+@pytest.mark.parametrize(
+    "data,expected",
+    [
+        (
+            pa.array([100, 200, 300], type=pa.decimal128(3)),
+            cudf.core.column.as_column(
+                [100, 200, 300], dtype=cudf.core.dtypes.Decimal64Dtype(3, 0)
+            ),
+        ),
+        (
+            pa.array([{"a": 1, "b": 3}, {"c": 2, "d": 4}]),
+            cudf.core.column.as_column([{"a": 1, "b": 3}, {"c": 2, "d": 4}]),
+        ),
+        (
+            pa.array([[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]]),
+            cudf.core.column.as_column(
+                [[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]]
+            ),
```

Comment:
This test gave a somewhat cryptic warning:

```
cudf/tests/test_column.py::test_as_column_arrow_array[data2-expected2]
  /home/charlesbluca/Documents/GitHub/compose/etc/conda/cuda_11.2/envs/rapids/lib/python3.8/site-packages/pandas/core/dtypes/missing.py:484: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    if np.any(np.asarray(left_value != right_value)):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

Is there something happening in the `as_column` operation here that would be causing the elementwise comparison to fail?

Reply:
`assert_eq` does it's final checking in a loop in python/pandas land. It's probably trying to compare two non-scalar objects ultimately and something weird is happening. I'd suggest calling `to_pandas` on the cuDF side of the comparison and seeing what the things that are being compared look like in the end.

Reply:
Checking out `to_pandas` shows the two objects do look the same:

```python
>>> cudf.Series(actual_column).to_pandas()
0       [[1, 2, 3], [4, 5, 6]]
1    [[7, 8, 9], [10, 11, 12]]
dtype: object
>>> cudf.Series(expected).to_pandas()
0       [[1, 2, 3], [4, 5, 6]]
1    [[7, 8, 9], [10, 11, 12]]
dtype: object
```

Could this warning just be a consequence of comparing list columns with nested lists?

Reply:
Probably. When cudf objects convert to pandas, it first goes through pyarrow. Pyarrow converts nested lists into nested arrays:
```
>>> arr = pa.array([[1, 2, 3]])
>>> arr
<pyarrow.lib.ListArray object at 0x7f43187e8a60>
[
  [
    1,
    2,
    3
  ]
]
>>> arr.to_pandas()
0    [1, 2, 3]
dtype: object
>>> arr.to_pandas().iloc[0]
array([1, 2, 3])
```